src/Search/Queue/WatchableQueue.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace eLife\Search\Queue;
+
+interface WatchableQueue
+{
+    /**
+     * Adds item to the queue.
+     *
+     * Mock: Add item to queue.
+     * SQS: This will set the queue item into the memory slot for re-processing.
+     */
+    public function enqueue(QueueItem $item) : bool;
+
+    /**
+     * Starts process of removing item.
+     *
+     * Mock: Move to separate "in progress" queue.
+     * SQS: this will change the timeout of the in-memory item.
+     */
+    public function dequeue(int $timeout = 30) : QueueItem;
+
+    /**
+     * Commits to removing item from queue, marks item as done and processed.
+     *
+     * Mock: Remove item completely.
+     * SQS: this will delete the item from the queue.
+     */
+    public function commit(QueueItem $item) : bool;
+
+    /**
+     * This will happen when an error happens, we release the item back into the queue.
+     *
+     * Mock: re-add to queue.
```

Comment:
I'm not sure what the memory slot is in the case of SQS. When a message is returned by the `receiveMessage` operation, it can be returned by `dequeue()` without changing the state of the `WatchableQueue` object. A "memory slot" may act as a buffer but what is the purpose?